webapp/src/views/projects/members/component/InviteDialog.tsx

```diff
@@ -0,0 +1,222 @@
+import { Formik, Field } from 'formik';
+import {
+  makeStyles,
+  Dialog,
+  DialogActions,
+  DialogTitle,
+  Button,
+  DialogContent,
+  TextField,
+  ButtonGroup,
+  Box,
+  Typography,
+} from '@material-ui/core';
+import { container } from 'tsyringe';
+import { useTranslate, T } from '@tolgee/react';
+import copy from 'copy-to-clipboard';
+import * as Yup from 'yup';
+
+import { useApiMutation } from 'tg.service/http/useQueryApi';
+import { useProject } from 'tg.hooks/useProject';
+import { PermissionsMenu } from 'tg.component/security/PermissionsMenu';
+import { LanguagePermissionsMenu } from 'tg.component/security/LanguagePermissionsMenu';
+import { components } from 'tg.service/apiSchema.generated';
+import LoadingButton from 'tg.component/common/form/LoadingButton';
+import { LINKS, PARAMS } from 'tg.constants/links';
+import { parseErrorResponse } from 'tg.fixtures/errorFIxtures';
+import { MessageService } from 'tg.service/MessageService';
+import { useMemo } from 'react';
+
+const messaging = container.resolve(MessageService);
+
+type computedPermissions = NonNullable<
+  components['schemas']['ProjectModel']['computedPermissions']
+>;
+
+const useStyles = makeStyles((theme) => ({
+  content: {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+    minHeight: 150,
+  },
+  permissions: {
+    display: 'flex',
+    gap: theme.spacing(1),
+  },
+}));
+
+type Props = {
+  open: boolean;
+  onClose: () => void;
+};
+
+export const InviteDialog: React.FC<Props> = ({ open, onClose }) => {
+  const classes = useStyles();
+  const t = useTranslate();
+  const project = useProject();
+  const invite = useApiMutation({
+    url: '/v2/projects/{projectId}/invite',
+    method: 'put',
+    invalidatePrefix: '/v2/projects/{projectId}/invitations',
+  });
+
+  const yupSchema = useMemo(
+    () =>
+      Yup.object({
```

Comment:
There is a file storing all Yup schemas, we should probably be consistent in this. It's there to make it easy to edit constraints for the fields globally.